integration/client/container_test.go

```diff
@@ -398,10 +399,24 @@ func TestContainerWait(t *testing.T) {
 	defer task.Delete(ctx)
 
 	ctx2, cancle2 := context.WithCancel(ctx)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	statusErrC := func() <-chan ExitStatus {
```

Comment:
Shouldn't this be the same as creating a new context with a 5 second timeout?

Reply:
They're not different. When the context ends due to a timeout, it's possible for both the statusErrC and the context (cancellation signal) to arrive simultaneously.

Expectation here is that the context cancel which then leads to an error on the status error channel